File: examples/java/patterns/pause-resume/callback-timeout.java
import software.amazon.lambda.durable.config.CallbackConfig;
import software.amazon.lambda.durable.config.WaitForCallbackConfig;

WaitForCallbackConfig config = WaitForCallbackConfig.builder()
    .callbackConfig(CallbackConfig.builder()
        .timeout(Duration.ofHours(24))
        .build())
    .build();

Approval outcome = context.waitForCallback(
    "wait-for-approval",
    Approval.class,
    (callbackId, ctx) -> approvalsService.request(input.orderId(), callbackId),
    config);

File: examples/python/patterns/pause-resume/callback-timeout.py
from aws_durable_execution_sdk_python.config import (
    Duration,
    WaitForCallbackConfig,
)

result = context.wait_for_callback(
    lambda callback_id, ctx: approvals_service.request(
        id=event["orderId"], callback_id=callback_id
    ),
    name="wait-for-approval",
    config=WaitForCallbackConfig(timeout=Duration.from_hours(24)),
)
